ql/src/test/queries/clientpositive/cbo_cte_materialization.q

```diff
@@ -0,0 +1,27 @@
+--! qt:dataset:src
+
+set hive.optimize.cte.materialize.threshold=1;
+set hive.optimize.cte.materialize.full.aggregate.only=false;
+
+EXPLAIN CBO
+WITH materialized_cte AS (
+  SELECT key, value FROM src WHERE key != '100'
+),
+another_materialized_cte AS (
+  SELECT key, value FROM src WHERE key != '100'
+)
+SELECT a.key, a.value, b.key, b.value
+FROM materialized_cte a
+JOIN another_materialized_cte b ON a.key = b.key
+ORDER BY a.key;
```

Comment:
Consider adding (or replacing this with) a traditional `EXPLAIN` where we can see the effect of `hive.optimize.cte.materialize.threshold` and the full plan for the materialized ctes.